src/index.spec.ts

```diff
@@ -1,12 +1,117 @@
+import fetch from 'node-fetch';
+
 import {
   isInteger,
   toLowerCase,
   removeDuplicatesFromArray,
   createRandomProduct,
   getStarWarsPlanets,
-  createProduct 
+  createProduct,
 } from './index';
 
+jest.mock('node-fetch');
+const { Response } = jest.requireActual('node-fetch');
+
 test('Testing time!', () => {
   console.log('welcome');
 });
+
+describe('isInteger', () => {
+  it('should be an integer', () => {
+    expect(isInteger(2)).toBe(true);
+  });
+
+  it('should be a string', () => {
+    expect(isInteger('hello')).toBe(false);
+  });
+});
+
+describe('toLowerCase', () => {
+  it('should fail if it is an empty word', () => {
+    expect(toLowerCase('')).toBe('Please provide a string');
+  });
+
+  it('should return a lower case word', () => {
+    expect(toLowerCase('HELLO')).toBe('hello');
+  });
+});
+
+describe('removeDuplicatesFromArray', () => {
+  it('should throw an error if arg is not an array', () => {
+    expect(() => {
+      removeDuplicatesFromArray(undefined as any);
+    }).toThrow('please provide an array of numbers or strings');
+  });
+
+  it('should return array unmodified if there is a single element', () => {
+    const shortArray = ['hello'];
+    expect(removeDuplicatesFromArray(shortArray)).toBe(shortArray);
+  });
+
+  it('should remove duplicates', () => {
+    const longArray = ['hello', 'bye', 'hello', 'good', 'hello'];
+    const result = removeDuplicatesFromArray(longArray);
+
+    expect(result.length).toBe(3);
+    expect(result).toEqual(['hello', 'bye', 'good']);
+  });
+});
+
+describe('createProduct', () => {
+  it('should throw an error when product is invalid (name length)', () => {
+    expect(() =>
+      createProduct({
+        name: 'p',
+        description: 'rechicken',
+        price: 12,
+        tags: ['food'],
+      }),
+    ).toThrow(Error);
+  });
+
+  it('should create a new product', () => {
+    const product = {
+      name: 'product',
+      description: 'rechicken',
+      price: 12,
+      tags: ['food'],
+    };
+
+    const result = createProduct(product);
+
+    expect(result).toHaveProperty('id');
```

Comment:
It is a good practice to include the expected property. In this scenario we don't know the 'id; value, since it was generated by faker, but we know its type, so we can enhance the test in the following way:

```
expect(result).toMatchObject({
      ...product,
      id: expect.any(Number),
    });
```